subsys/net/lib/aws_fota/src/aws_fota_json.c

```diff
@@ -132,92 +25,116 @@ static void strncpy_nullterm(char *dst, const char *src, size_t maxlen)
 	}
 }
 
-int aws_fota_parse_UpdateJobExecution_rsp(char *update_rsp_document,
-					  size_t payload_len, char *status)
+int aws_fota_parse_UpdateJobExecution_rsp(const char *update_rsp_document,
+					  size_t payload_len, char *status_buf)
 {
-	if (update_rsp_document == NULL || status == NULL) {
+	if (update_rsp_document == NULL || status_buf == NULL) {
 		return -EINVAL;
 	}
 
-	struct update_rsp_obj rsp;
-
-	int ret = json_obj_parse(update_rsp_document, payload_len,
-			update_job_exec_stat_rsp_descr,
-			ARRAY_SIZE(update_job_exec_stat_rsp_descr), &rsp);
+	cJSON * update_response = cJSON_Parse(update_rsp_document);
 
-	if (ret < 0) {
-		return ret;
+	if (update_response == NULL) {
+		cJSON_Delete(update_response);
+		return -ENODATA;
 	}
 
-	/* Check if the status field(1st field) of the object has been parsed */
-	if (ret & BIT(UPDATE_JOB_EXECUTION_STATUS_DECODED_BIT)) {
-		strncpy_nullterm(status, rsp.status, STATUS_MAX_LEN);
+	cJSON * status = cJSON_GetObjectItemCaseSensitive(update_response,
+							  "status");
+	if (cJSON_IsString(status) && status->valuestring != NULL) {
+		strncpy_nullterm(status_buf, status->valuestring,
+				 STATUS_MAX_LEN);
 	} else {
+		cJSON_Delete(update_response);
 		return -ENODATA;
 	}
 
+	cJSON_Delete(update_response);
 	return 0;
 }
 
-int aws_fota_parse_DescribeJobExecution_rsp(char *job_document,
+int aws_fota_parse_DescribeJobExecution_rsp(const char *job_document,
 					   u32_t payload_len,
 					   char *job_id_buf,
 					   char *hostname_buf,
 					   char *file_path_buf,
-					   int *version_number)
+					   int *execution_version_number)
 {
 	if (job_document == NULL
 	    || job_id_buf == NULL
 	    || hostname_buf == NULL
 	    || file_path_buf == NULL
-	    || version_number == NULL) {
+	    || execution_version_number == NULL) {
 		return -EINVAL;
 	}
 
-	struct desc_job_execution_obj root_obj;
-	struct job_document_obj *job_doc_obj;
-	struct execution_obj *exec_obj;
-
-	int ret = json_obj_parse(job_document, payload_len,
-				 desc_job_execution_obj_descr,
-				 ARRAY_SIZE(desc_job_execution_obj_descr),
-				 &root_obj);
+	cJSON * json_data = cJSON_Parse(job_document);
 
-	if (ret < 0) {
-		return ret;
+	if (json_data == NULL) {
+		const char *err = cJSON_GetErrorPtr();
+		/* LOG_ERR("Unable to parse Job Execution err before: %s", err); */
+		cJSON_Delete(json_data);
+		return -ENODATA;
 	}
 
-	if (!(ret & BIT(EXECUTION_OBJ_DECODED_BIT))) {
-		/* No execution object in JSON */
+	cJSON * execution = cJSON_GetObjectItemCaseSensitive(json_data,
+							    "execution");
+	if (execution == NULL) {
 		return 0;
```

Comment:
No need to delete `json_data` here?

Reply:
Would have to `delete` it there. Changed the code to use `goto cleanup`

Reply:
Tell me if you want me to change back or use a `do while` instead of goto.

Reply:
`goto` is fine IMO